src/icp_canister_backend/tests/rewards_tests.rs

```diff
@@ -90,73 +86,310 @@ fn test_reward_rate_increase_decrease_during_episodes() {
     );
 }
 
+
 #[test]
 fn test_reward_distribution_middle_and_final() {
     let (pic, canister_id, ledger_id) = setup();
     let user = Principal::from_text("xkbqi-2qaaa-aaaah-qbpqq-cai").unwrap();
-    let deposit_amount = Nat::from(100_000_000u64);
+    let deposit_amount = Nat::from(100_000_000u64); // 1 BTC
     let reward_amount = Nat::from(50_000_000u64);
-
+ 
     let stakable_episode = get_stakable_episode(&pic, canister_id, 7);
-    create_deposit(
-        &pic,
-        canister_id,
-        ledger_id,
-        user,
-        deposit_amount.clone(),
-        stakable_episode,
-    );
-
+    create_deposit(&pic, canister_id, ledger_id, user, deposit_amount.clone(), stakable_episode);
+    
     let reward_time = pic.get_time().as_nanos_since_unix_epoch() / 1_000_000_000;
-
+    
     reward_pool(&pic, canister_id, ledger_id, user, reward_amount.clone())
         .expect("Reward pool should succeed");
 
     let last_reward_episode = (reward_time + EPISODE_DURATION * 12) / EPISODE_DURATION;
     let exact_reward_duration = (last_reward_episode + 1) * EPISODE_DURATION - reward_time;
-
-    //Middle  rewards distribution
+    
+      //Middle  rewards distribution
     let half_duration = exact_reward_duration / 2;
     advance_time(&pic, half_duration);
-
+    
     let middle_rewards = pic
-        .query_call(
-            canister_id,
-            user,
-            "get_deposits_rewards",
-            encode_args((vec![0u64],)).unwrap(),
-        )
+        .query_call(canister_id, user, "get_deposits_rewards", encode_args((vec![0u64],)).unwrap())
         .map(|r| decode_one::<Nat>(&r).unwrap())
         .expect("Failed to get middle rewards");
-
+    
+    
     let expected_middle = reward_amount.clone() / Nat::from(2u64);
-
-    assert_with_error!(
-        &middle_rewards,
-        &expected_middle,
-        &ALLOWED_ERROR,
-        "Middle reward distribution"
-    );
-
+   
+    assert_with_error!(&middle_rewards, &expected_middle, &ALLOWED_ERROR, "Middle reward distribution");
+    
+    
     let remaining_duration = exact_reward_duration - half_duration;
     advance_time(&pic, remaining_duration);
-
+    
     let final_rewards = pic
-        .query_call(
-            canister_id,
-            user,
-            "get_deposits_rewards",
-            encode_args((vec![0u64],)).unwrap(),
-        )
+        .query_call(canister_id, user, "get_deposits_rewards", encode_args((vec![0u64],)).unwrap())
         .map(|r| decode_one::<Nat>(&r).unwrap())
         .expect("Failed to get final rewards");
-
+    
     // final reward distribution
     let expected_final = reward_amount.clone();
-    assert_with_error!(
-        &final_rewards,
-        &expected_final,
-        &ALLOWED_ERROR,
-        "Full reward distribution"
-    );
+    assert_with_error!(&final_rewards, &expected_final, &ALLOWED_ERROR, "Full reward distribution");
```

Comment:
Also, add withdraw_rewards function call.
and call it double time in the end to ensure, that it doesn't allow to withdraw reward several times.
And decrease the initial reward to x2, to be sure that it doesn't fail because of not enough funds on the canister.